api/edge/v1alpha1/types.go

```diff
@@ -171,6 +173,125 @@ type PlacementList struct {
+	// Standard object metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// `spec` explicitly describes a desired binding between workloads and Locations.
+	// It reflects the resolution of a Placement's selectors.
+	Spec PlacementDecisionSpec `json:"spec,omitempty"`
+
+	// `status` describes the status of the process of binding workloads to Locations.
+	// +optional
+	Status PlacementDecisionStatus `json:"status,omitempty"`
+}
+
+// PlacementDecisionSpec holds a list of objects and a list of destinations which are the resolution
+// of a Placement's `what` and `where`: what objects to propagate and to where.
+// All objects present in this spec are propagated to all destinations present.
+type PlacementDecisionSpec struct {
+	// `Workload` is a collection of namespaced-scoped objects and a collection of cluster-scoped objects to be propagated to destination clusters.
+	Workload DownsyncObjectsReferences `json:"workload,omitempty"`
+
+	// `destinations` is a list of cluster-identifiers that the objects should be propagated to.
+	Destinations []Destination `json:"destinations,omitempty"`
+}
+
+// DownsyncObjectsReferences explicitly defines the objects to be down-synced.
+// The ClusterScope list defines the cluster-scope objects, while NamespacedObjects packs individual objects
+// identifiable by namespace & name.
+type DownsyncObjectsReferences struct {
+	// `clusterScope` holds a list of individual cluster-scoped objects
+	// to downsync, organized by resource.
+	// Remember that a "resource" is a kind/type/sort of objects,
+	// not an individual object.
+	// +optional
+	ClusterScope []ClusterScopeDownsyncObject `json:"clusterScope,omitempty"`
+
+	// `NamespaceScope` matches if and only if at least one member matches.
+	// +optional
+	NamespaceScope []NamespaceScopeDownsyncObjects `json:"namespaceScope,omitempty"`
+}
+
+// NamespaceScopeDownsyncObjects matches some objects of one particular namespaced object.
+type NamespaceScopeDownsyncObjects struct {
+	// GroupVersionResource holds the API group, API version and resource name.
+	metav1.GroupVersionResource `json:",inline"`
+
+	// `objectsByNamespace` matches by namespace and name.
+	// An object matches the list if and only if the object matches at least one member of the list.
+	// Thus, no object matches the empty list.
+	// +optional
+	ObjectsByNamespace []NamespaceAndNames `json:"objectsByNamespace,omitempty"`
+}
+
+// NamespaceAndNames identifies some objects of an implied resource that is namespaced.
+// The objects are all in the same namespace.
+type NamespaceAndNames struct {
+	// `namespace` identifies the namespace
+	Namespace string `json:"namespace"`
+
+	// `names` holds the names of the objects that match.
+	// Empty list means none of them.
+	// +optional
+	Names []string `json:"names,omitempty"`
+}
+
+type ClusterScopeDownsyncObject struct {
```

Comment:
Since this identifies generally multiple objects, I would put a plural in the type name. "ClusterScopeDownsyncObjects".

Reply:
done